src/qt/pivx/settings/settingsinformationwidget.cpp

```diff
@@ -136,6 +140,9 @@ void SettingsInformationWidget::loadClientModel(){
 
         setNumBlocks(clientModel->getNumBlocks());
         connect(clientModel, &ClientModel::numBlocksChanged, this, &SettingsInformationWidget::setNumBlocks);
+
+        setMasternodeCount(clientModel->getMasternodeCountString());
```

Comment:
Hmm, `getMasternodeCountString` implementation is calling to `mnodeman.CountNetworks` which is not locking the masternodeman mutex when it should as it's accessing a thread shared object.

Same as in many other places, `CountNetworks` will need a review.

Reply:
Just realized that we have a timer locking `cs_main` to update the MNs count in clientModel.

You can connect to the signal `strMasternodesChanged` to get the `cachedMasternodeCountString`. 

`getMasternodeCountString` should be a private method and not public one.

And we shouldn't lock `cs_main` in the `updateMnTimer` polling method and only lock the masternodes manager `cs`

Reply:
going to leave most of this for a separate PR as much of it is not directly related to the addition of these 2 new lines to the settings area, but deeper in the back end. I have removed the (un-necessary) call to `setMasternodeCount` in `loadClientModel` though.